packages/server/src/tests/header/headerSelection.ts

```diff
@@ -107,91 +91,96 @@ const supportersTest: HeaderTest = {
     ],
 };
 
-const supportersTestUS: HeaderTest = {
-    name: 'header-supporter__USEOY',
-    audience: 'AllExistingSupporters',
-    variants: [
-        {
-            name: 'control',
-            modulePathBuilder: usEOYHeader.endpointPathBuilder,
-            content: {
-                heading: 'Thank you for supporting us',
-                subheading: '',
-                primaryCta: {
-                    url:
-                        'https://support.theguardian.com/us/contribute?selected-contribution-type=ONE_OFF',
-                    text: 'Make an extra contribution',
-                },
-            },
-            mobileContent: {
-                heading: 'Thank you',
-                subheading: '',
-                primaryCta: {
-                    url:
-                        'https://support.theguardian.com/us/contribute?selected-contribution-type=ONE_OFF',
-                    text: 'Contribute again',
-                },
-            },
-        },
-    ],
-};
-
-const usEoyPeriodStart = new Date(2021, 10, 22);
-const usEoyPeriodEnd = new Date(2022, 1, 1);
+const existingSupportersAndEveryone = ['Everyone', 'AllExistingSupporters'];
 
-const isInUsEoyPeriod = (date: Date): boolean => {
-    return isAfter(date, usEoyPeriodStart) && isBefore(date, usEoyPeriodEnd);
-};
+const selectBestTest = (
+    targeting: HeaderTargeting,
+    allTests: HeaderTest[],
+): HeaderTestSelection => {
+    // If there is a need to return other specific hardcoded tests/variants, include required logic to identify, and required return, here
 
-const hasNotContributedDuringUsEoyPeriod = (lastOneOffContributionDate?: string): boolean => {
-    if (!lastOneOffContributionDate) {
-        return true;
+    // In case the allTests array is empty, return a hardcoded test/variant
+    if (!allTests.length) {
+        return selectHardcodedTest(targeting);
     }
 
-    return isBefore(new Date(Date.parse(lastOneOffContributionDate)), usEoyPeriodStart);
-};
+    // Cascade through allTests to match supplied targeting data to the first test that meets requirements - if no test meets requirements, return an appropriate hardcoded test/variant
+    let selectedTest;
+
+    const { showSupportMessaging, countryCode } = targeting;
+
+    const countryGroupId = countryCodeToCountryGroupId(countryCode.toUpperCase());
+
+    for (let i = 0, iz = allTests.length; i < iz; i++) {
```

Comment:
I think [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) is much simpler and more maintainable than a for loop. It removes the need for variables. E.g.

```
const selectedTest: HeaderTest | undefined = allTests.find(test => {
  // return true if user can be in test
});
```

Reply:
Done (I think). Will need to be carefully smoke tested to make sure the function returns what we believe it should be returning.